src/edgepi/dac/edgepi_dac.py

```diff
@@ -50,26 +65,43 @@ def __init__(self):
         self.gpio = EdgePiGPIO()
 
         self.__dac_power_state = {
+            DACChannel.AOUT8.value: PowerMode.NORMAL.value,
             DACChannel.AOUT7.value: PowerMode.NORMAL.value,
             DACChannel.AOUT6.value: PowerMode.NORMAL.value,
             DACChannel.AOUT5.value: PowerMode.NORMAL.value,
             DACChannel.AOUT4.value: PowerMode.NORMAL.value,
             DACChannel.AOUT3.value: PowerMode.NORMAL.value,
             DACChannel.AOUT2.value: PowerMode.NORMAL.value,
             DACChannel.AOUT1.value: PowerMode.NORMAL.value,
-            DACChannel.AOUT0.value: PowerMode.NORMAL.value,
         }
 
     def __send_to_gpio_pins(self, analog_out: int, voltage: float):
         ao_pin = self.__analog_out_pin_map[analog_out].value
-
+        do_pin = self.__analog_to_digital_pin_map[analog_out].value
         if voltage > 0:
             self.gpio.set_pin_state(ao_pin)
+            self.gpio.clear_pin_state(do_pin)
+            if analog_out in (DACChannel.AOUT1.value, DACChannel.AOUT2.value):
+                self.__dac_switching_logic(analog_out)
         elif voltage == 0:
             self.gpio.clear_pin_state(ao_pin)
+            if analog_out in (DACChannel.AOUT2.value,DACChannel.AOUT1.value):
+                self.__dac_switching_logic(analog_out)
```

Comment:
can you use lists here instead? the `(` and `)` may be mistaken for a tuple. it does the same thing but, if you are listing out possible values, it's generally better to use a list

Reply:
df7eb18d92d9516015533bb2808e871da631df3a